deepmd/pt/model/descriptor/sezm.py

```
@@ -568,7 +616,13 @@ def __init__(
 
         # === L/M schedules ===
         self._init_lm_schedules(lmax, n_blocks, l_schedule, mmax, m_schedule)
+        self.kmax = int(kmax)
+        if self.kmax < 0:
+            raise ValueError("`kmax` must be non-negative")
+        if self.kmax > int(lmax):
+            raise ValueError("`kmax` must be <= `lmax`")
         self.ebed_dims = [get_so3_dim_of_lmax(l) for l in self.l_schedule]
```

Ruff (0.15.15)

[error] Ambiguous variable name: `l`

(E741)